File: primesieve_info.pas
program primesieve_info;
{$IFDEF FPC}
{$MODE Delphi}
{$ENDIF}

{$APPTYPE Console}

uses SysUtils, primesieve;

begin
  WriteLn('PrimeSieve information');
  WriteLn('----------------------');  
  WriteLn(Format('Library version: %s', [primesieve_version()]));
  WriteLn(Format('API version: %s', [_PRIMESIEVE_VERSION]));
  WriteLn(Format('_PRIMESIEVE_ERROR constant: %x', [_PRIMESIEVE_ERROR]));  
  WriteLn(Format('Largest valid stop number: %x', [primesieve_get_max_stop()]));
  WriteLn(Format('Current sieve size in KiB: %d', [primesieve_get_sieve_size()]));  
  WriteLn(Format('Current number of threads: %d', [primesieve_get_num_threads()]));    
  WriteLn(Format('SizeOf primesieve_iterator record: %d', [SizeOf(primesieve_iterator)]))
end.

File: primesieve_info.c
#include <inttypes.h>
#include <stdio.h>
#include <primesieve.h>

int main() {
  printf("PrimeSieve information\n");
  printf("----------------------\n");
  printf("Library version: %s\n", primesieve_version());
  printf("API version: %s\n", PRIMESIEVE_VERSION);
  printf("PRIMESIEVE_ERROR constant: %" PRIX64 "\n", PRIMESIEVE_ERROR);
  printf("Largest valid stop number: %" PRIX64 "\n", primesieve_get_max_stop());
  printf("Current sieve size in KiB: %d\n", primesieve_get_sieve_size());
  printf("Current number of threads: %d\n", primesieve_get_num_threads());
  printf("SizeOf primesieve_iterator record: %ld\n", sizeof(primesieve_iterator));
  return 0;
}
